src/modules/Lan.rb

```diff
@@ -214,51 +214,45 @@ def isAnyInterfaceDown
       down
     end
 
+    # Checks local configuration if IPv6 is allowed
+    #
+    # return [Boolean] true when IPv6 is allowed
     def readIPv6
       @ipv6 = true
 
-      methods =
-        #         "module" : $[
-        #                 "filelist" : ["ipv6", "50-ipv6.conf"],
-        #                 "filepath" : "/etc/modprobe.d/",
-        #                 "regexp"   : "^[[:space:]]*(install ipv6 /bin/true)"
-        #         ]
-        {
-          "builtin" => {
-            "filelist" => ["sysctl.conf"],
-            "filepath" => "/etc/",
-            "regexp"   => "^[[:space:]]*(net.ipv6.conf.all.disable_ipv6)[[:space:]]*=[[:space:]]*1"
-          }
+      methods = {
+        "builtin" => {
+          "filelist" => ["sysctl.conf"],
+          "filepath" => "/etc",
+          "regexp"   => /^[[:space:]]*(net.ipv6.conf.all.disable_ipv6)[[:space:]]*=[[:space:]]*1/
         }
+      }
 
-      Builtins.foreach(methods) do |which, method|
-        filelist = Ops.get_list(method, "filelist", [])
-        filepath = Ops.get_string(method, "filepath", "")
-        regexp = Ops.get_string(method, "regexp", "")
-        Builtins.foreach(filelist) do |file|
-          filename = Builtins.sformat("%1/%2", filepath, file)
-          if FileUtils.Exists(filename)
-            Builtins.foreach(
-              Builtins.splitstring(
-                Convert.to_string(SCR.Read(path(".target.string"), filename)),
-                "\n"
-              )
-            ) do |row|
-              if Ops.greater_than(
-                Builtins.size(
-                  Builtins.regexptokenize(String.CutBlanks(row), regexp)
-                ),
-                0
-              )
-                Builtins.y2milestone("IPv6 is disabled by '%1' method.", which)
-                @ipv6 = false
-              end
-            end
+      methods.each do |_, method|
+        filelist = method["filelist"] || []
+        filepath = method["filepath"] || ""
+        regexp = method["regexp"] || ""
+
+        filelist.each do |file|
+          filename = "#{filepath}/#{file}"
+          next if !FileUtils.Exists(filename)
+
+          begin
+            lines = (SCR.Read(path(".target.string"), filename) || []).split("\n")
+          rescue ArgumentError => e
+            log.error("readIPv6: an error when parsing #{filename}")
+            log.error(e.message)
+
+            return false
```

Comment:
I remembered my final decision in this (I've tried several approaches, so I've a little confused myself ;-)

```true / false``` from this method means if the configuration of IPv6 was / was not read successfully.  That's also reason why I decided to handle exception here. And that's also reason why I didn't remove setting ```@ipv6``` from this method.

Based on ```true / false``` result upper layers can decide if it is critical error for them or not. When running in GUI, user is asked (in an upper layer which uses that). When running in non-interactive mode (AY), then it should fail.